src/cargo/util/progress.rs

```diff
@@ -116,6 +146,14 @@ impl<'cfg> State<'cfg> {
         string.push_str("]");
         string.push_str(&stats);
 
+        let avail_msg_len = self.max_width - self.width;
+        if avail_msg_len >= msg.len() + 3 {
+            string.push_str(&msg);
+        } else if avail_msg_len >= 4 {
+            string.push_str(&msg[..(avail_msg_len - 3)]);
```

Comment:
Crate names can be non-ascii:

```
~/tmp
λ cargo new "привет-мир"
     Created binary (application) `привет-мир` project

~/tmp
λ cd привет-мир

~/tmp/привет-мир master*
λ cargo run
   Compiling привет-мир v0.1.0 (file:///home/matklad/tmp/%D0%BF%D1%80%D0%B8%D0%B2%D0%B5%D1%82-%D0%BC%D0%B8%D1%80)
    Finished dev [unoptimized + debuginfo] target(s) in 0.99 secs
     Running `target/debug/привет-мир`
Hello, world!
```

So I think this *might* fail at run time with utf8-boundary error :-) If we fix this, might be a good idea to write a unit-test for this formatting function as well.

Reply:
Ah thanks. Since this is a UI component we should limit by the [display width](https://crates.io/crates/unicode-width), which counts by grapheme cluster and each cluster's width can be 1 or 2 spaces (half width or full width) long 💀. Since your Cargo.lock already depends on `unicode-width` [via `clap`](https://github.com/kbknapp/clap-rs/blob/94a16e6ddf3360181cb424b8d029ad5a93a979fe/Cargo.toml#L28), I'll just add this dependency.